salt/metalk8s/orchestrate/deploy_new_node.sls

```diff
@@ -1,47 +1,61 @@
-{%- if pillar['bootstrap_id'] %}
-{%-   set control_plane_ips = salt.saltutil.runner('mine.get', tgt=pillar['bootstrap_id'], fun='control_plane_ip') %}
-{%- else %}
-{%-   set control_plane_ips = {} %}
-{%- endif %}
+{%- set control_plane_ip = salt.saltutil.runner(
+    'mine.get',
+    tgt=pillar['bootstrap_id'],
+    fun='control_plane_ip'
+)[pillar['bootstrap_id']] %}
 
-{%- if pillar['bootstrap_id'] in control_plane_ips.keys() and control_plane_ips[pillar['bootstrap_id']] %}
-{%-   set control_plane_ip = control_plane_ips[pillar['bootstrap_id']] %}
+{%- set pillar_data = {
+    'repo': {
+        'host': control_plane_ip
+    },
+    'registry_ip': control_plane_ip
+} %}
+
+{%- if 'metalk8s' in pillar
+        and 'nodes' in pillar.metalk8s
+        and pillar.bootstrap_id in pillar.metalk8s.nodes
+        and pillar.metalk8s.nodes[pillar.node_name].version is not none %}
+  {%- set version = pillar.metalk8s.nodes[pillar.bootstrap_id].version %}
 {%- else %}
-{%-   set control_plane_ip = 'localhost' %}
+  {%- set _, version = saltenv.split('-', 1) %}
 {%- endif %}
 
-
+{%- if pillar['node_name'] not in salt['saltutil.runner']('manage.up') %}
 Deploy salt-minion on a new node:
   salt.state:
     - ssh: true
     - roster: kubernetes
     - tgt: {{ pillar['node_name'] }}
-    - saltenv: {{ saltenv }}
+    - saltenv: metalk8s-{{ version }}
     - sls:
       - metalk8s.roles.minion
-    - pillar:
-        repo:
-          host: {{ control_plane_ip }}
-        salt:
-          master:
-            host: {{ control_plane_ip }}
+    - pillar: {{ pillar_data | tojson }}
 
 Accept key:
```

Comment:
Can we do this through a `wheel` module instead of going to the minion to then `crictl` something?